models/CommentsByArticleIdModel.js

```diff
@@ -0,0 +1,24 @@
+const db = require('../db/connection.js')
+
+const getCommentsByArticleId = (article_id) => {
+
+    const articleId = article_id
+
+    return db.query(`
+    SELECT * FROM comments
+    WHERE article_id = $1
+    ORDER BY created_at DESC
+    `, [articleId]).then((result) => {
+        const data = result.rows
+
+        if (data && data.length === 0) {
+            return Promise.reject('article not found')
```

Comment:
This promise.reject should return an object with a status and a msg. 
Not sure how your error handling has worked without it...
Your error handler expects an err.status and an err.msg later on - these should come from your rejected promise here. So in this case it would be a 404 and a msg of article not found